app/lib/account/models.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:gcloud/db.dart' as db;
+
+/// User data model.
+@db.Kind(name: 'User', idType: db.IdType.String)
+class User extends db.ExpandoModel {
+  String get userId => id as String;
+
+  @db.StringProperty()
+  String authProvider;
+
+  @db.StringProperty()
+  String authUserId;
+
+  @db.StringProperty()
+  String email;
+
+  @db.StringListProperty()
+  List<String> historicalEmails = <String>[];
+
+  @db.BoolProperty()
+  bool hideEmail = false;
+
+  @db.DateTimeProperty()
+  DateTime created;
+
+  @db.DateTimeProperty()
+  DateTime updated;
+
+  @db.DateTimeProperty()
+  DateTime lastAccess;
```

Comment:
Let's rename this to `lastLogin`, as access could imply the last time this record was accessed.

Reply:
hmm, thinking...what if we change this to `lastLoginDate`, and then reduce precision to just the date... That way we only need to update it once per day per user...  Multiple requests the same on the same date won't cause updates...

This seems like the kind of thing that could cause write contention. And having the last date is probably fine for most users..

Reply:
removed